pyiceberg/manifest.py

```diff
@@ -573,6 +573,17 @@ class ManifestFile(Record):
     def __init__(self, *data: Any, **named_data: Any) -> None:
         super().__init__(*data, **{"struct": MANIFEST_LIST_FILE_STRUCTS[DEFAULT_READ_VERSION], **named_data})
 
+    def __eq__(self, other: Any) -> bool:
+        """Return the equality of two instances of the ManifestFile class."""
+        if not isinstance(other, ManifestFile):
+            return False
+        else:
+            return self.manifest_path == other.manifest_path
```

Comment:
nit: I always do this in a single statement:
```suggestion
        return self.manifest_path == other.manifest_path if isinstance(other, ManifestFile) else False
```